aggregate/billing_functions/seqr.py

```diff
@@ -0,0 +1,593 @@
+            bq.ScalarQueryParameter('end', 'STRING', str(end)),
+            bq.ScalarQueryParameter('project', 'STRING', 'seqr-308602'),
+        ]
+    )
+
+    df = (
+        utils.bigquery_client.query(_query, job_config=job_config)
+        .result()
+        .to_dataframe()
+    )
+    json_obj = json.loads(df.to_json(orient='records'))
+
+    entries_by_id = {}
+    result = 0
+    param_map, current_date = None, None
+    for obj in json_obj:
+
+        if obj['cost'] == 0:
+            # come on now
+            continue
+        usage_start_time = datetime.fromtimestamp(int(obj['usage_start_time'] / 1000))
+        del obj['billing_account_id']
+        del obj['project']
+        if current_date is None or usage_start_time > current_date:
+            # use and abuse the
+            param_map = get_ratios_from_date(usage_start_time, prop_map)
+
+        labels = obj['labels']
+
+        for dataset, ratio in param_map.items():
+
+            new_entry = deepcopy(obj)
+
+            new_entry['topic'] = dataset
+            new_entry['service']['id'] = SERVICE_ID
+            new_entry['labels'] = [*labels, {'key': 'proportion', 'value': ratio}]
+            new_entry['cost'] *= ratio
+
+            dates = ['usage_start_time', 'usage_end_time', 'export_time']
+            for k in dates:
+                new_entry[k] = utils.to_bq_time(
+                    datetime.fromtimestamp(int(obj[k] / 1000))
+                )
+
+            new_entry['id'] = '-'.join(
+                [SERVICE_ID, dataset, billing_row_to_key(new_entry)]
+            )
+
+            if new_entry['id'] in entries_by_id:
+                if new_entry != entries_by_id[new_entry['id']]:
+                    logger.warning(
+                        f'WARNING: duplicate entry {new_entry["id"]} with diff values'
+                    )
+                continue
+
+            entries_by_id[new_entry['id']] = new_entry
+
+        if len(entries_by_id) > 10000:
+
+            # insert all entries here
+            credit_entries = {
+                e['id']: e for e in utils.get_seqr_credits(entries_by_id.values())
+            }
+            entries_by_id.update(credit_entries)
+
+            if len(entries_by_id) != 2 * len(credit_entries):
+                logger.warning('Mismatched length of credits')
+
+            result += utils.insert_new_rows_in_table(
+                table=utils.GCP_AGGREGATE_DEST_TABLE,
+                obj=entries_by_id.values(),
+                dry_run=dry_run,
+            )
+            entries_by_id = {}
+
+    if entries_by_id:
+        entries_by_id.update(
+            {e['id']: e for e in utils.get_seqr_credits(entries_by_id.values())}
+        )
+
+        result += utils.insert_new_rows_in_table(
+            table=utils.GCP_AGGREGATE_DEST_TABLE,
+            obj=entries_by_id.values(),
+            dry_run=dry_run,
+        )
+
+    return result
+
+
+def billing_row_to_key(row) -> str:
```

Comment:
Add type annotation to row, and probably update the method name to suggest it generates from a dictionary